app/models/events/pull_request.rb

```diff
@@ -16,23 +16,28 @@
 #  updated_at :datetime         not null
 #  github_id  :bigint           not null
 #  node_id    :string           not null
+#  owner_id   :bigint
 #  project_id :bigint           not null
 #
 # Indexes
 #
 #  index_pull_requests_on_github_id   (github_id) UNIQUE
+#  index_pull_requests_on_owner_id    (owner_id)
 #  index_pull_requests_on_project_id  (project_id)
 #  index_pull_requests_on_state       (state)
 #
 # Foreign Keys
 #
+#  fk_rails_...  (owner_id => users.id)
 #  fk_rails_...  (project_id => projects.id)
 #
 module Events
   class PullRequest < ApplicationRecord
     enum state: { open: 'open', closed: 'closed' }
 
     belongs_to :project, inverse_of: :pull_requests
+    belongs_to :owner, class_name: 'User', foreign_key: :owner_id,
```

Comment:
question, what happens with existing records that don't have a `owner_id`? Are they going to be considered "invalid" for the rest of their lives?

Reply:
At some point we will need to re process all the events again because of the amount of changes after the deployment. But yes, you have a point here, the migration will fail because of existing records without the owner_id, at first we should have this field with a default on the db side, and delete this constraint when we have re processed all the events.